metals/src/main/scala/scala/meta/metals/mtags/ScalaMtags.scala

```diff
@@ -42,8 +42,30 @@ object ScalaMtags {
           case t: Defn.Object => term(t.name, OBJECT); continue()
           case t: Defn.Type => tpe(t.name, TYPE); stop()
           case t: Decl.Type => tpe(t.name, TYPE); stop()
-          case t: Defn.Def => term(t.name, DEF); stop()
-          case t: Decl.Def => term(t.name, DEF); stop()
+          case t: Defn.Def => 
+            for {
+              params <- t.paramss
+              tpes = params.flatMap(_.decltpe)
+              names = tpes.map{
+                case d: Type.Name => d
+              }
+            } withOwner() {
+              val params = names.mkString("(", ",", ")")
+              super.method(t.name, params, DEF)
+            }
+            stop()
+          case t: Decl.Def => 
+            for {
+              params <- t.paramss
+              tpes = params.flatMap(_.decltpe)
+              names = tpes.map{
+                case d: Type.Name => d
+              }
+            } withOwner() {
+              val params = names.mkString("(", ",", ")")
+              super.method(t.name, params, DEF)
+            }
+            stop()
```

Comment:
Unless I'm mistaken, the two implementations for `Decl/Defn` are identical. This could probably be made an helper that takes `Seq[Term.Param]` in input. Something like

```scala
case t: Defn.Def => def(t.paramss)
case t: Decl.Def => def(t.paramss)
```

Reply:
I cannot find a common type between Defn.Def and Decl.Def that has `paramss` :/ So either I repeat the logic or I do a weird pattern match only on those in the helper. Any suggestions?

Reply:
If you write an helper like

```scala

Reply:
You probably need the name too, so it would be

```
def defTerm(name: Term.Name, paramss: Seq[Term.Param]) = {
  // ...
}
```

or similar (I suspect you can't call a `def` `"def"` :-P)